### ISSUE TITLE
[Assistive tech and device support] UI components initiate a change of context when they receive user input. (11.13.1)

### ISSUE THREAD

**Issue by shiragoodman** [department-of-veterans-affairs/va.gov-team#55849]
#### VFS team name
VA Online Scheduling (VAOS)

#### VFS product name
VA Online Scheduling

#### VFS feature name
Appointment List

#### Point of Contact/Reviewers
Brian DeConinck - @briandeconinck - Accessibility

*For more information on how to interpret this ticket, please refer to the [Anatomy of a Staging Review issue ticket](https://depo-platform-documentation.scrollhelp.site/collaboration-cycle/Anatomy-of-a-Staging-Review-Issue-ticket.2060320997.html) guidance on Platform Website.

---

### Platform Issue
UI components initiate a change of context when they receive user input.

### Issue Details
On the Past appointments page, there's an option to filter by date range with a `select` dropdown. Choosing an option in the dropdown triggers the JS action that filters the appointments.

Changing context just from selecting an option in the dropdown is an unexpected pattern not used elsewhere on VA.gov and may be disorienting to some users. The [documentation around WCAG 3.2.2](https://www.w3.org/WAI/WCAG21/Understanding/on-input.html) explores this in more detail.

### VA.gov Experience Standard
[Category Number 11, Issue Number 13](https://depo-platform-documentation.scrollhelp.site/collaboration-cycle/VA.gov-experience-standards.1683980311.html)

### Other References
WCAG SC 3.2.2_A

### Platform Recommendation
Add a button (eg. "Filter") alongside the `select`, and use the button to trigger the JS action. See also [WCAG Technique G80](https://www.w3.org/WAI/WCAG21/Techniques/general/G80).

---

### VFS Guidance

- Close the ticket when the issue has been resolved or validated by your Product Owner
- If your team has additional questions or needs Platform help validating the issue, please comment on the ticket
- Some feedback provided may be out of scope for your iteration of the product, however, Platform's OCTO leadership has stated that all identified issues need to be documented and it is still your responsibility to resolve the issue.
- If you do not believe that this Staging Review issue ticket is the responsibility of your team, comment below providing an explanation and who you believe is responsible. Please tag the Point of Contact/Reviewers. Governance team will research and will follow up.

**Comment by outerpress** [comment]
@briandeconinck @humancompanion-usds do you know of any good examples of select list filter patterns that we should borrow from? The Facility Locator search is pretty close.

**Comment by ldelacosta** [comment]
We will explore this recommendation post launch for the Appointment List feature. We have created a new [ticket](https://github.com/department-of-veterans-affairs/va.gov-team/issues/57143) for this work. cc: @JRRoof

**Comment by shiragoodman** [comment]
thanks @ldelacosta . Per our guidance above, we request that this ticket remain open until the issue is resolved. Thanks for your understanding.

**Comment by humancompanion-usds** [comment]
To answer @outerpress's question - No, I don't think select is a good component to use for filtering. I don't love the [date range picker in USWDS](https://designsystem.digital.gov/components/date-range-picker/) but, that's an option (perhaps hidden in a "Filters" accordion or other drawer)

**Comment by JRRoof** [comment]
How do we insure we get more timely responses to comments and questions raised in these tickets?  2 months for a reply is not an acceptable time frame with our product owner.

**Comment by shiragoodman** [comment]
@JRRoof - Slack or Office Hours are the best option for getting timely responses, as things can fall through the cracks with Github. In the future, I encourage you and your team to either join the appropriate [Office Hours](https://depo-platform-documentation.scrollhelp.site/support/office-hours) or reach out on #vfs-platform-support Slack channel.
